go/integrityblock/integrityblock.go

```diff
@@ -228,3 +228,35 @@ func WebBundleHasIntegrityBlock(bundleFile io.ReadSeeker) (bool, error) {
 
 	return bytes.Compare(IntegrityBlockMagic, possibleMagic) == 0, nil
 }
+
+// GetIntegrityBlockWithNewSignature contains the main logic for generating the new signature and
+// prepending the integrity block's signature stack with a new integrity signature object.
+func GetIntegrityBlockWithNewSignature(integrityBlock *IntegrityBlock, ed25519privKey ed25519.PrivateKey, webBundleHash []byte, signatureAttributes map[string][]byte) (*IntegrityBlock, error) {
```

Comment:
The `IntegrityBlock` return value is redundant, because this function always returns the `IntegrityBlock` in the first parameter.

Since this function mutates the given `IntegrityBlock`, I'd make it a method of `IntegrityBlock` type. The function signature would be:
```Go
func (integrityBlock *IntegrityBlock) AddNewSignature(ed25519privKey ed25519.PrivateKey, webBundleHash []byte, signatureAttributes map[string][]byte) error
```